src/bus/driver.c

```diff
@@ -1349,6 +1349,47 @@ static int driver_method_remove_match(Peer *peer, CDVar *in_v, uint32_t serial,
         return 0;
 }
 
+int driver_reload_config_completed(Bus *bus, uint64_t sender_id, uint32_t reply_serial) {
+        Peer *sender;
+        int r;
+
+        sender = peer_registry_find_peer(&bus->peers, sender_id);
+        if (sender) {
+                _c_cleanup_(c_dvar_deinit) CDVar var = C_DVAR_INIT;
+
+                c_dvar_begin_write(&var, driver_type_out_unit, 1);
+                c_dvar_write(&var, "(");
+                driver_write_reply_header(&var, sender, reply_serial, driver_type_out_unit);
+                c_dvar_write(&var, "()");
+
+                r = driver_send_reply(sender, &var, reply_serial);
+                if (r)
+                        return error_trace(r);
+        }
```

Comment:
So you went with the single-object approach, meaning we never cleanup the pending reload-requests on disconnect, right?

Just for clarity: They're accounted on the uid, and pending on our trusted launcher. Hence, there is no exploit possible, IIUC.

Reply:
Correct. Though for the record, I don't see how cleaning up on disconnect would affect this regardless, as accounting is done per-uid and not per-connection.